federation/pkg/federation-controller/replicaset/replicasetcontroller.go

```diff
@@ -580,24 +586,33 @@ func (frsc *ReplicaSetController) reconcileReplicaSet(key string) (reconciliatio
 		} else {
 			currentLrs := lrsObj.(*extensionsv1.ReplicaSet)
 			// Update existing replica set, if needed.
-			if !fedutil.ObjectMetaAndSpecEquivalent(lrs, currentLrs) {
-				frsc.eventRecorder.Eventf(frs, api.EventTypeNormal, "UpdateInCluster",
-					"Updating replicaset in cluster %s", clusterName)
+			if !isPuppet(frs) {
+				if !fedutil.ObjectMetaAndSpecEquivalent(lrs, currentLrs) {
+					frsc.eventRecorder.Eventf(frs, api.EventTypeNormal, "UpdateInCluster",
+						"Updating replicaset in cluster %s", clusterName)
+
+					operations = append(operations, fedutil.FederatedOperation{
+						Type:        fedutil.OperationTypeUpdate,
+						Obj:         lrs,
+						ClusterName: clusterName,
+					})
+				}
+			}
 
-				operations = append(operations, fedutil.FederatedOperation{
-					Type:        fedutil.OperationTypeUpdate,
-					Obj:         lrs,
-					ClusterName: clusterName,
-				})
+			// If selector and pod template matches the spec then include it in the stats.
+			if reflect.DeepEqual(frs.Spec.Selector, currentLrs.Spec.Selector) &&
+				reflect.DeepEqual(frs.Spec.Template, currentLrs.Spec.Template) {
+
+				// This sum is only applied on puppets.
+				localSpecReplicaSum += *currentLrs.Spec.Replicas
+				updateStatus(&fedStatus, &currentLrs.Status)
 			}
-			fedStatus.Replicas += currentLrs.Status.Replicas
-			fedStatus.FullyLabeledReplicas += currentLrs.Status.FullyLabeledReplicas
-			fedStatus.ReadyReplicas += currentLrs.Status.ReadyReplicas
-			fedStatus.AvailableReplicas += currentLrs.Status.AvailableReplicas
 		}
 	}
-	if fedStatus.Replicas != frs.Status.Replicas || fedStatus.FullyLabeledReplicas != frs.Status.FullyLabeledReplicas ||
-		fedStatus.ReadyReplicas != frs.Status.ReadyReplicas || fedStatus.AvailableReplicas != frs.Status.AvailableReplicas {
+	if hasStatusChanged(&frs.Status, &fedStatus) || (isPuppet(frs) && *frs.Spec.Replicas != localSpecReplicaSum) {
+		if isPuppet(frs) {
+			frs.Spec.Replicas = &localSpecReplicaSum
```

Comment:
We will need another call to Update the spec. UpdateStatus below will only update status